docs/shared_bindings_matrix.py

```diff
@@ -117,168 +79,71 @@ def build_module_map():
-                            base[module]["default_value"] not in [
-                                "None",
-                                "CIRCUITPY_DEFAULT_BUILD"
-                            ]):
-                                check_dependent_modules[module] = base[module]["default_value"]
-                else:
-                    board_is_excluded = find_module.group(1) == "0"
-
-                if board_is_excluded:
-                    if board_chip in base[module]["excluded"]:
-                        base[module]["excluded"][board_chip].append(entry.name)
-                    else:
-                        base[module]["excluded"][board_chip] = [entry.name]
-
-            for module in check_dependent_modules:
-                depend_results = set()
-
-                parents = check_dependent_modules[module].split("CIRCUITPY_")
-                parents = [item.strip(", ").lower() for item in parents if item]
-
-                for parent in parents:
-                    if parent in base:
-                        if (board_chip in base[parent]["excluded"] and
-                            entry.name in base[parent]["excluded"][board_chip]):
-                                depend_results.add(False)
-                        else:
-                            depend_results.add(True)
-
-                # only exclude the module if there were zero parents enabled
-                # as determined by the 'depend_results' set.
-                if not any(depend_results):
-                    if board_chip in base[module]["excluded"]:
-                        base[module]["excluded"][board_chip].append(entry.name)
-                    else:
-                        base[module]["excluded"][board_chip] = [entry.name]
-
-    #print(json.dumps(base, indent=2))
     return base
 
+def get_settings_from_makefile(port_dir, board_name):
+    """ Invoke make in a mode which prints the database, then parse it for
+        settings.
 
-def support_matrix_excluded_boards():
-    """ Compiles a list of available modules, and which board definitions
-        do not include them.
+        This means that the effect of all Makefile directives is taken
+        into account, without having to re-encode the logic that sets them
+        in this script, something that has proved error-prone
     """
-    base = build_module_map()
 
-    return get_excluded_boards(base)
+    status, contents = subprocess.getstatusoutput(f"make -C {port_dir} BOARD={board_name} -qp print-CC")
+    # Make signals errors with exit status 2; 0 and 1 are "non-error" statuses
+    if status not in (0, 1):
+        raise RuntimeError(f'Invoking make exited with {status}')
+    if isinstance(contents, bytes):
+        contents = contents.decode('utf-8', errors='replace')
+    settings = {}
+    for line in contents.split('\n'):
+        m = re.match(r'^([A-Z][A-Z0-9_]*) = (.*)$', line)
+        if m:
+            settings[m.group(1)] = m.group(2)
+    return settings
```

Comment:
Nothing wrong here, really. But, when I was testing, I had to change up the call to figure out what exactly was breaking. Using `subprocess.run` allows for some better error messaging, which I think would prove handy since this is largely machine-run.
 This was my local version:
```py
contents = subprocess.run(
        ["make", "-C", port_dir, f"BOARD={board_name}", "-qp", "print-CC"],
        encoding="utf-8",
        errors="replace",
        stdout=subprocess.PIPE,
        stderr=subprocess.PIPE
    )
    # Make signals errors with exit status 2; 0 and 1 are "non-error" statuses
    if contents.returncode not in (0, 1):
        error_msg = (
            f"Invoking '{' '.join(contents.args)}' exited with "
            f"{contents.returncode}: {contents.stderr}"
        )
        raise RuntimeError(error_msg)

    settings = {}
    for line in contents.stdout.split('\n'):
        m = re.match(r'^([A-Z][A-Z0-9_]*) = (.*)$', line)
        if m:
            settings[m.group(1)] = m.group(2)

    return settings
```

Reply:
Good suggestion, thanks.

I actually had the list-style invocation at first, but it failed for me.  If it works now I'll add it to the PR.